### ISSUE TITLE
TLS connection failed to open

### ISSUE THREAD

**Issue by gael-ian** [djfarrelly/MailDev#234]
Hello,

MailDev worked fine for a while but I'm now experiencing systematic crashes as soon as I try to send an email.

```bash
▶ node -v
v10.11.0
▶ yarn -v
1.10.1
▶ yarn run maildev -vo
yarn run v1.10.1
$ /[project path]/node_modules/.bin/maildev -v
Temporary directory created at /tmp/maildev/20023
MailDev webapp running at http://0.0.0.0:1080
MailDev SMTP Server running at 0.0.0.0:1025
Could not start mail server on undefined:undefined
Port already in use or insufficient rights to bind port
Recieved shutdown signal, shutting down now...
Done in 135.68s.
```

First, no tab will open in my browser unless I launch maildev with `sudo`. If I open it manualy, the app seems to run correctly. Second, it crashs as soon as I try to send an email.

I tried to find the origin of the problem, but I'm not very familiar with the Node environment. As far as I could go, it seems something wrong happen in `SMTPConnection.init()` and the connection can not be initialized with the socket.

1025 and 1080 ports are free and I bet the SMTP socket port is fairly random. It's not a permission problem as, even when launched with sudo, the connection crash the same way (just the web UI opens in my browser).

All version of MailDev seems affected (I tested each one from 1.0.0-rc3 to 0.130).

**Comment by gael-ian** [comment]
Hi,

I dug a little deeper this morning. First, I cloned this repo, runned `npm run dev` unit tests and saw mails popping in the web interface. I discover you run MailDev without TLS for development, so I tried to do the same in my app.

*A non-TLS MailDev SMTP server works fine.*

Before this issue gets solved, a simple workaround is to disable TLS, either from the app point of view or from MailDev point of vue. For example, in Rails (my use case), TLS is enabled by default so the correct configuration to use is:

```ruby
config.action_mailer.delivery_method = :smtp
config.action_mailer.smtp_settings = {
  address: "localhost",
  port: 1025,
  enable_starttls_auto: false
}
```

Alternatively, MailDev can be runned without TLS support with `maildev --hide-extensions STARTTLS`.
